### DISCUSSION TITLE
please add the variant named Paradigm (Paradigm Chess30)

### DISCUSSION THREAD

**Discussion by emanuelesaiu** [gbtami/pychess-variants#1410] [Ideas]
Is Paradigm Chess30 still missing?
Paradigm Chess30 was invented in South Africa in 2020, and can be played on chess.com
What's more important, it's a very good and professional-style variant.
https://paradigmchess30.com
https://www.chessvariants.com/rules/paradigm-chess30

It is supported by Fairy-Stockfish!

As for the set of pieces (graphics), it needs a winged bishop: exactly the one that can be selected for Miguel Illescas's dragon chess in PyChess. (But Illescas's "dragon" is the usual bishop-knight hybrid, and I would like it could be displayed as such -- can't it? The unique piece of Paradigm Chess30, instead, is a bishop-xiangqihorse hybrid.)

**Comment by gbtami** [comment]
Please read my general answer about adding new variants here https://discord.com/channels/634298688663191582/634298688663191584/1226913584521674792

**Reply by emanuelesaiu** [reply]
«
We already added dragon chess (arcbishop in the pocket) and have S-chess
There are almost endless variation for this type of variants (adding one or two new pieces only)
I think we will not add more in the near future for sure
»

Thank you for your link to the general answer. I get your point and think you are right, as far as useless / non-existent / highly repetitive variants are concerned!

But let me spend some more lines for Paradigm Chess30. ( https://paradigmchess30.com )

- This variant was created in 2020 and is not very famous, but has huge potential. It's more than good enough, in my opinion. Does it bring anything new compared to the other variants you already have? Sure it does. The resemblance to Capablanca/Seirawan/Illescas variants is only superficial! Just read the rules carefully, or watch videos etc. (see below).

- The bishop-xiangqihorse hybrid (aka Paradigm dragons, or Paradigm dragon bishops) is unique to Paradigm Chess, as far as I know. (Each player has two. Normal bishops are absent.)

- It's extremely sober, entirely chess-like, with no redundancy at all: normal 8x8 chessboard, and it's NOT a drop variant. No pocket, nothing in pockets.

- It has some established user base, at least in South African chess clubs (in person), and online -- especially on chess.com, which is not a minor platform.

- It is supported by Fairy-Stockfish, as already said.

- There is a Youtube channel @paradigm_chess30 with some dozen videos for a lot of hours in total (but, well, not many views). That includes videos such as "Introduction to Paradigm Chess30 - Part 1 (Theory)".

- Chess.com supports Paradigm Chess30! Ok, around 60% of chess.com variants are rubbish and/or childish amusement, but this is not the case.
Currently, Paradigm Chess30 is on https://www.chess.com/variants in the sub-section "Featured". This sub-section consists of ten variants, among which are:
. Seirawan Chess (aka S-Chess),
. Gothic Chess (one of the "classical" Capablanca 10x8 variations),
. so-called Chaturanga (that is, their tentative reconstruction of proto-chess).
Paradigm players on chess.com are not legion: true. Not many, but some dozens of them are smart and fast players, with good or very good ratings. It's usually easy to find decent opponents in a few minutes. (I have personal experience of it, as a good chess amateur and a Paradigm beginner myself.)

(If the peculiar kind of starting position randomization is somewhat of a problem, I suppose we could be satisfied with a standard-position-only Paradigm Chess. The usual PyChess form "Paste the FEN text here" will be there, right? That's important!)
(There are 30 possibilities, as the name Chess30 suggests -- and they are NOT a subset of the famous 960, since non-colour-bound Paradigm dragons replace the bishops.)

Thank you again for your attention. Regards,

Emanuele Saiu
(Italy)

**Reply by gbtami** [reply]
Starting position randomization is trivial so it is not a real problem. 
Seems this variant has at least the same potential similar to Miguel Illescas's dragon chess.
I will ask the other pychess devs opinion about adding it to the site in the next months.
Thx for the additional info!

**Reply by emanuelesaiu** [reply]
Since randomization is trivial, I suggest that (in case **Paradigm Chess30** is added) a subvariant with full Fischer-style randomization is added too.
**Fischer randomization, and Fischer castling**.
This subvariant can be named **Paradigm Chess1320**: that's because Paradigm Chess has **no colour-bound pieces**, and the Fischer Random constraint on opposite-coloured bishops would make no sense.

(About Paradigm Chess with full Fischer-style randomization, that is Paradigm Chess1320, see section §5 in my web page [https://sites.google.com/site/emanuelesaiu/chess-paradigm](https://sites.google.com/site/emanuelesaiu/chess-paradigm) )

Thank you again!

Emanuele Saiu

**Reply by emanuelesaiu** [reply]
I have found out that **Paradigm Chess30 is already available** in one PyChess **fork**! It's at a herokuapp.com domain. I have tried it, it seems to work alright.
(It's the standard "30" version, of course: no further randomization.)
But, I don't know, probably it's been made by some of the people who work for the official PyChess, so I'm not giving you any useful information. Is that the case?

Emanuele Saiu

**Reply by gbtami** [reply]
Yes. You can find it here https://github.com/autocorr/pychess-variants/tree/alternates

**Reply by gbtami** [reply]
You can talk to autocorr on pychess Discord https://discord.gg/aPs8RKr variant-workshop channel about it.
